packages/grafana-ui/src/components/Forms/Checkbox.tsx

```diff
@@ -124,6 +131,39 @@ export const getCheckboxStyles = stylesFactory((theme: GrafanaTheme2) => {
         }
       }
     `,
+
+    inputIndeterminate: css`
+      &[aria-checked='mixed'] + span {
+        border: none;
+        background: ${theme.colors.primary.main};
+
+        &:hover {
+          background: ${theme.colors.primary.shade};
+        }
+
+        &:after {
+          content: '';
+          position: absolute;
+          z-index: 2;
+          left: 3px;
+          right: 3px;
+          top: calc(50% - 1.5px);
+          height: 3px;
+          border: solid ${theme.colors.primary.contrastText};
```

Comment:
@JoaoSilvaGrafana If I get rid of this the `indeterminate state` looks good, otherwise it looks like the screenshots I added to my comments.

Reply:
Firefox! :shakes-fist:
Thanks for checking Laura 😄 I will see how to solve this, that line is there solving another edge case problem...

Reply:
Happy to help! 😄